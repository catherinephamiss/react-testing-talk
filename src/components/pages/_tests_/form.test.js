import React from "react";
import Form from "../form";
import renderer from "react-test-renderer";
import expectExport from "expect";

/*
 * Test are just chunks of code that are used to test code that manages our business logic.
 * Testing can be integrated into your deployment pipeline to make sure you don't ship any breaking bugs
 */

//Let's start by taking a snapshot of the form component. A snapshot is a look at how the component renders and looks for unexpected changes.
//snap shot are an test a component's implementation, which, can change. They are easily updated and truthfully, not that useful.
it("renders correctly", () => {
  const tree = renderer.create(<Form />).toJSON;
  expectExport(tree).toMatchSnapshot();
});
