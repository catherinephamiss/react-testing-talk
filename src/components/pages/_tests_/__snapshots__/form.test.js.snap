// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="form-holder"
>
  <div
    class="container"
  >
    <h2>
      Log in
    </h2>
    <div>
      We haven't tracked you in awhile
    </div>
  </div>
  <form>
    <div
      class="input-holder"
    >
      <label
        for="email"
      >
        Email:
      </label>
      <input
        data-testid="email"
        id="email"
        placeholder="you@you.com"
        required=""
        type="email"
        value=""
      />
    </div>
    <div
      class="input-holder"
    >
      <label
        for="password"
      >
        Password:
      </label>
      <input
        data-testid="password"
        id="password"
        placeholder="your password"
        required=""
        type="password"
        value=""
      />
    </div>
    <div
      class="container"
    >
      <button
        type="submit"
      >
        Let us track you
      </button>
    </div>
  </form>
</div>
`;
